primitives/core/src/lib.rs

```diff
@@ -80,6 +80,7 @@ pub use changes_trie::{ChangesTrieConfiguration, ChangesTrieConfigurationRange};
 #[cfg(feature = "full_crypto")]
 pub use crypto::{DeriveJunction, Pair, Public};
 
+pub use crypto::NodePublicKey;
```

Comment:
Nit, since I'm not a huge fan of re-exports, but why not just use `sp_core::crypto::NodePublicKey` on the call site?

Reply:
The intention here is to let user (runtime developer) not care about the inner layer. I guess that's part of designing a library. But I'm not very sure whether `crypto` is  something can be un-cared by user, `crypto` sounds like a utility to me, and `NodePublicKey` more about logic/business. That's why I decided to re-export it.